src/test/java/cloud/klasse/backendbusiness/BackendBusinessApplicationTests.java

```diff
@@ -1,17 +1,131 @@
 package cloud.klasse.backendbusiness;
 
+import cloud.klasse.backendbusiness.student.CreateStudentModel;
+import cloud.klasse.backendbusiness.user.User;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.client.RootUriTemplateHandler;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
 import org.springframework.test.context.TestPropertySource;
 
-@SpringBootTest
-@TestPropertySource(
-        properties = "spring.jpa.hibernate.ddl-auto=update"
-)
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestPropertySource(properties = "spring.jpa.hibernate.ddl-auto=update")
 class BackendBusinessApplicationTests {
 
+    @LocalServerPort
+    int serverPort;
+
+    private final Base64.Encoder base64Encoder = Base64.getEncoder().withoutPadding();
+
+    private final TestRestTemplate restTemplate = new TestRestTemplate(TestRestTemplate.HttpClientOption.ENABLE_COOKIES);
+
+    @BeforeAll
+    void setUp() {
+        restTemplate.setUriTemplateHandler(new RootUriTemplateHandler(String.format("http://localhost:%s", serverPort)));
+    }
+
     @Test
-    void contextLoads() {
+    void securityIntegrationTest() {
+        // Attempt to access protected resource without access token => Should be rejected
+        final var unauthorizedResponse = restTemplate.getForEntity("/user/1", User.class);
+        assertThat(unauthorizedResponse.getStatusCode()).isEqualTo(HttpStatus.FORBIDDEN);
+
+        // Attempt to create user without access token => Should be accepted
+        final var createUserModel = new CreateStudentModel();
+        createUserModel.setUserName("Klaus");
+        createUserModel.setNickName("klaus");
+        createUserModel.setPassword("It's super-cereal");
+        final var createUserResponse = restTemplate.postForEntity("/user", createUserModel, User.class);
```

Comment:
I thought we’ve renamed the api endpoint to /register

Reply:
Any reason for that? We can rename it to register, but that would not be RESTful API / best practice.
Is your intention to make clear that there is more logic and other API calls required?

Reply:
No. We should use restful API. But Andreas and I were not shure how to deal with login and register. In the Rest request file there is currently the path register

Reply:
Sorry. I mean the create-user http file

Reply:
The re-rename is basically the result of the merge of the refactoring of the User/Student/Teacher API of the.... (Hmmm. Nope, I'm done with "of the"s).

Reply:
The integration test is also working (actual HTTP request against an active server on localhost)

Reply:
can we merge this? Would like to submit file API, but would like to merge first.